File: examples/csharp/l3-resources/atlas-basic.cs
using System;
using Amazon.CDK;
using Constructs;
using MongoDB.AWSCDKResourcesMongoDBAtlas;
using MongoDB.AWSCDKResourcesMongoDBAtlas;
using ProjectProps = MongoDB.AWSCDKResourcesMongoDBAtlas.ProjectProps;


namespace CdkTestAppCsharp
{
    public class CdkTestAppCsharpStack : Stack
    {
        internal CdkTestAppCsharpStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            var profile = (string) this.Node.TryGetContext("profile");
            var orgId = (string) this.Node.TryGetContext("orgId");
            var ipAddr = (string) this.Node.TryGetContext("ipAddr");
            var ipComment = (string) this.Node.TryGetContext("ipComment");
            var region = (string) this.Node.TryGetContext("region");

             var regionConfig = new AdvancedRegionConfig
                     {
                         AnalyticsSpecs = new Specs { NodeCount = 1, InstanceSize = "M10", EbsVolumeType = "STANDARD" },
                         ElectableSpecs = new Specs { NodeCount = 3, InstanceSize = "M10", EbsVolumeType = "STANDARD" },
                         Priority = 7,
                         RegionName = region
                     };
             var regionConfigs = new AdvancedRegionConfig[1] { regionConfig };

             var replicationSpec = new AdvancedReplicationSpec { AdvancedRegionConfigs = regionConfigs, NumShards = 1 };
             var replicationSpecs = new AdvancedReplicationSpec[1] { replicationSpec };

             var accessListDef = new AccessListDefinition { IpAddress = ipAddr, Comment = ipComment };
             var ipAccessListProps = new IpAccessListProps { AccessList = new AccessListDefinition[1] { accessListDef } };

             var atlasBasic = new AtlasBasic(this, "AtlasBasic-l3", new AtlasBasicProps
                    {
                        ClusterProps = new ClusterProps { ReplicationSpecs = replicationSpecs },
                        ProjectProps = new ProjectProps { OrgId = orgId },
                        IpAccessListProps = ipAccessListProps,
                        Profile = profile,
                    });
        }
    }
}

File: examples/golang/l3-resources/atlas-basic.go
package l3_resources

import (
	mdb "github.com/mongodb/awscdk-resources-mongodbatlas-go/awscdkresourcesmongodbatlas"
)

const (
	ORG_ID     = "ORG_ID_HERE"
	PROFILE    = "PROFILE_HERE"
	PROJECT_ID = "PROJECT_ID_HERE"
	API_KEY    = "API_KEY_HERE"
	IP_ADDRESS = "IP_ADDRESS_HERE"
	IP_COMMENT = "COMMENT_HERE"
	REGION     = "REGION_HERE"
)

type CdkTestAppGoStackProps struct {
	awscdk.StackProps
}

func NewCdkTestAppGoStack(scope constructs.Construct, id string, props *CdkTestAppGoStackProps) awscdk.Stack {
	var sprops awscdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := awscdk.NewStack(scope, &id, &sprops)

	profile := getContextProp(stack.Node(), PROFILE)
	orgId := getContextProp(stack.Node(), ORG_ID)
	ipAddr := getContextProp(stack.Node(), IP_ADDRESS)
	ipComment := getContextProp(stack.Node(), IP_COMMENT)
	region := getContextProp(stack.Node(), REGION)

	regionConfig := &mdb.AdvancedRegionConfig{
		AnalyticsSpecs: &mdb.Specs{
			EbsVolumeType: jsii.String("STANDARD"),
			InstanceSize:  jsii.String("M10"),
			NodeCount:     jsii.Number(1),
		},
		ElectableSpecs: &mdb.Specs{
			EbsVolumeType: jsii.String("STANDARD"),
			InstanceSize:  jsii.String("M10"),
			NodeCount:     jsii.Number(3),
		},
		Priority:   jsii.Number(7),
		RegionName: jsii.String(region),
	}

	regionConfigs := &[]*mdb.AdvancedRegionConfig{regionConfig}
	replicationSpecs := &[]*mdb.AdvancedReplicationSpec{
		{
			AdvancedRegionConfigs: regionConfigs,
			NumShards:             jsii.Number(1),
		},
	}

	ipAccessListProps := mdb.IpAccessListProps{
		AccessList: &[]*mdb.AccessListDefinition{
			{
				Comment:   jsii.String(ipComment),
				IpAddress: jsii.String(ipAddr),
			},
		},
	}

	mdb.NewAtlasBasic(stack, jsii.String("AtlasBasic-go"), &mdb.AtlasBasicProps{
		ClusterProps: &mdb.ClusterProps{
			ReplicationSpecs: replicationSpecs,
		},
		ProjectProps: &mdb.ProjectProps{
			OrgId: jsii.String(orgId),
		},
		IpAccessListProps: &ipAccessListProps,
		Profile:           jsii.String(profile),
	})

	return stack
}

func getContextProp(pNode constructs.Node, propName string) string {
	ctxValue := pNode.TryGetContext(jsii.String(propName))
	if v, ok := ctxValue.(string); ok {
		return v
	}
	return ""
}
